core/src/main/scala/kafka/network/RequestChannel.scala

```diff
@@ -512,6 +512,44 @@ class RequestChannel(val queueSize: Int,
 
 }
 
+class RequestBreakdownMetrics(val name: String) extends KafkaMetricsGroup {
+  val HistogramMetricType = "histogram"
+  val DurationDataType = "duration"
+  val tags = Map("request" -> name)
+  // This is to make writing MBean query easier
+  val tagsForHistograms = tags ++ Map("metricType" -> HistogramMetricType)
+  val tagsForDurationHistograms = tagsForHistograms ++ Map("metricType" -> HistogramMetricType, "dataType" -> DurationDataType)
```

Comment:
Where do the metricType and dataType tags get used?

Reply:
This is for writing mbean query in the kafka-server sensor easier.
The sensor glue code needs to determine whether it's histogram/gauge/etc.

For existing MBeans, such as request rates, they need to be listed one by one in the sensor code.
But for this time on the new metric `RequestBreakdownStats`, I think it would be easier if we put generic types in, so that in the future, we don't need to modify too much on kafka-server to add new sensor.

Reply:
Changed the implementation.  Now the MBean is consistent with the previous convention.

However, for observability I still added a field for kafka-server to inspect what are the histogram names.  

```scala
val DurationHistogramMetricNames: mutable.Set[String]
```